### ISSUE TITLE
Get-OktaUser Returns an Empty Object

### ISSUE THREAD

**Issue by mattmcnabb** [mattmcnabb/Okta#3]
Consider this command:

```
PS C:\> Get-OktaUser -Identity mcnabb

FirstName             :
LastName              :
Login                 :
Email                 :
Activated             :
Created               :
Credentials           :
Id                    :
LastLogin             :
LastUpdated           :
PasswordChanged       :
Profile               :
Status                :
StatusChanged         :
TransitioningToStatus :
```

This should simply return null if no user is found, but instead it returns an empty user object. This does not appear to happen when using the `-Filter` parameter, so likely an issue with output from the `GetUser()` static method.

**Comment by mattmcnabb** [comment]
It appears that the core of the issue is that the Get* methods on the user class are failing silently. This prevents `Connect-Okta` from properly testing the connection details, and will allow functions to continue even if the API connection fails. Need to make sure that these methods always throw an exception.

**Comment by mattmcnabb** [comment]
I've investigated error handling in both the C# side and the calling PowerShell and decided the best bet is in the parent caller in PowerShell. The problem is that we need to call `$Task.Wait()` to handle the errors. The best solution in this case I believe is to implement a generic `Wait-Task` function to handle the resultant error or success output. With this we can pass the original API exception to the calling function where it can be wrapped with a contextual exception message.

**Comment by mattmcnabb** [comment]
Fix is implemented - once this can be tested on all platforms it's ready to close.

**Comment by mattmcnabb** [comment]
tested on Windows PowerShell, PowerShell Core for Windows and Linux. Merged in #5

**Comment by mattmcnabb** [comment]
Reopening since I forgot to actually fix `Get-OktaUser!` 8b6783d9696465d750d7d8b13227db663c607161 built the capability to fix this, but only actually fixed the issue in `Connect-Okta. `Get-OktaUser` will need to implement `Wait-Task` to call this issue complete.
